appservice/src/appSettings/downloadAppSettings.ts

```diff
@@ -0,0 +1,49 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { WebSiteManagementModels } from "@azure/arm-appservice";
+import * as fse from 'fs-extra';
+import * as vscode from 'vscode';
+import { ext } from "../extensionVariables";
+import { localize } from "../localize";
+import { AppSettingsTreeItem } from "../tree/AppSettingsTreeItem";
+import { confirmOverwriteSettings } from "./confirmOverwriteSettings";
```

Comment:
Why does "Download App Settings" use the shared app service package, but "Create Function" uses the functions extension API? How are you deciding which feature uses which method?

Reply:
Well, downloadAppSetting has a lot of shared code between App Service, Functions, and SWA.  However-- this code isn't _quite_ generic enough as it is since App Service's .env files don't rely on a JSON object.

Do you think it'd make more sense to expand the Functions API to include the download/upload app settings logic?  I really hadn't considered that.

Reply:
I'm not convinced `downloadAppSettings` could be shared with App Service - would have to see how you make that generic. If it only works for Functions/SWA, that's where my question comes in. I don't want to start sticking a bunch of Functions-specific features in this package just for SWA.

Reply:
Yeah, I think it makes sense to leave OverwriteSettings in here (since it's pretty generic) but, I think I like the idea of doing all the appSetting stuff through the Functions API since it's technically all Functions under the hood.  Are you good with that?

Reply:
Yeah I think so